### PR TITLE
#1858 - Add currency formatting for payment amount

### PR DESCRIPTION
**Pull request by MackHalliday**
# Description

issue #1858 

- Update the formatting of payment amount to have commas. Ex. "1000.00" is now "1,000.00" 
- Added related testing
- Fix test failing with Runtime Error `tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py::test_it_update_dynamo_item_is_processed_updates_properly`

## Type of change

Please check the relevant option(s).
- [x] Internal
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (quick fix for an urgent bug or issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

- Deployed to Perf, set test notifications to `is_processed` is False, wait for task to run, check on phone that commas appear. (thanks @EvanParish )
image
image

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column
- [x] I have added a bullet for this work to the Engineering Key Wins slide for review

### CODE REVIEW

File: app/integrations/comp_and_pen/scheduled_message_helpers.py [hunk 1]
```diff
@@ -149,7 +149,8 @@ def send_comp_and_pen_sms(
         for item in comp_and_pen_messages:
             vaprofile_id = str(item.get('vaprofile_id'))
             participant_id = item.get('participant_id')
-            payment_amount = str(item.get('paymentAmount'))
+            # Format payment amount as str with appropriate commas
+            payment_amount = '{:,.2f}'.format((item.get('paymentAmount')))
```
[the remark this answers is not in the document]
Comment (k-macmillan) [reply app/integrations/comp_and_pen/scheduled_message_helpers.py:153]: https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Process/style_guide.md

[the remark this answers is not in the document]
Comment (MackHalliday) [reply app/integrations/comp_and_pen/scheduled_message_helpers.py:153]: man - I would have never gotten this.

[the remark this answers is not in the document]
Comment (MackHalliday) [reply app/integrations/comp_and_pen/scheduled_message_helpers.py]: Updated.

File: app/integrations/comp_and_pen/scheduled_message_helpers.py [hunk 2]
```diff
@@ -149,7 +149,8 @@ def send_comp_and_pen_sms(
         for item in comp_and_pen_messages:
             vaprofile_id = str(item.get('vaprofile_id'))
             participant_id = item.get('participant_id')
-            payment_amount = str(item.get('paymentAmount'))
+            # Format payment amount as str with appropriate commas
+            payment_amount = f'{item.get("paymentAmount", 0):0,.2f}'
```
[the remark this answers is not in the document]
Comment (MackHalliday) [reply app/integrations/comp_and_pen/scheduled_message_helpers.py:153]: The validation of paymentAmount is a concern handled upstream. This method's responsibility is to send the SMS notifications. Therefore, we should not need to test if paymentAmount is None ({'paymentAmount':None}), as this validation is already addressed by the upstream processes. I would advocate for keeping the responsibilities of each method separate.

File: tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py [hunk 3]
```diff
@@ -165,3 +167,106 @@ def test_ut_send_scheduled_comp_and_pen_sms_calls_send_notification_with_recipie
...
+
+
+def test_ut_send_scheduled_comp_and_pen_sms_payment_amount_key_does_not_exist(
+    mocker,
+    msg_helper,
+    dynamodb_mock,
+    sample_service,
+    sample_template,
+):
+    dynamo_data = [
```
Comment (MackHalliday) [inline tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py:235]: Removed `paymentAmount` from  `dynamo_data` dict. 

[As addressed above - edge will not happen. ](https://github.com/department-of-veterans-affairs/notification-api/pull/1887/files#r1693611884)

File: tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py [hunk 4]
```diff
@@ -165,3 +167,106 @@ def test_ut_send_scheduled_comp_and_pen_sms_calls_send_notification_with_recipie
...
+@pytest.mark.parametrize(
+    'amount, formatted_amount',
+    [
+        (1123.05, '1,123.05'),
+        (1000.00, '1,000.00'),
+        (10000.00, '10,000.00'),
+        (1234567.89, '1,234,567.89'),
+        (50.5, '50.50'),
+        (0.5, '0.50'),
+        (0.0, '0.00'),
```
Comment (MackHalliday) [inline tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py:181]: I tried adding  (None, '0.00') to this parameterization. It fails with the error `TypeError: unsupported format string passed to NoneType.__format__`.

`f'{item.get("paymentAmount", 0):0,.2f}'` sets the default to `0` only if the `payment amount` key does not exist. If the `paymentAmount` key has a value of None ({`paymentAmount`: `None`}), then this code will try to format `None`. 

We could add an `if/else` statement - something like `if item.get(`paymentAmount`) is None... else`. But validating the `paymentAmount` exists is handled upstream in multiple places. Adding another if/else statement would create a redundant code. 

[See doc on `.get()`](https://www.w3schools.com/python/ref_dictionary_get.asp#:~:text=the%20value%20from-,value,return%20if%20the%20specified%20key%20does%20not%20exist.%0ADefault%20value%20None,-More%20Examples)

Comment (MackHalliday) [reply tests/app/integrations/comp_and_pen/test_scheduled_message_helpers.py:181]: [I did add a test for if the `paymentAmount` key does not exists at all. ](https://github.com/department-of-veterans-affairs/notification-api/pull/1887/files#r1693702398)

**Review by cris-oddball — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]
